sklearn/decomposition/_dict_learning.py

```diff
@@ -1152,6 +1033,182 @@ def dict_learning_online(
         return dictionary
 
 
+@validate_params(
+    {
+        "X": ["array-like"],
+        "n_components": [Interval(Integral, 1, None, closed="left"), None],
+        "alpha": [Interval(Real, 0, None, closed="left")],
+        "max_iter": [Interval(Integral, 0, None, closed="left")],
+        "tol": [Interval(Real, 0, None, closed="left")],
+        "method": [StrOptions({"lars", "cd"})],
+        "n_jobs": [Integral, None],
+        "dict_init": [np.ndarray, None],
+        "code_init": [np.ndarray, None],
+        "callback": [callable, None],
+        "verbose": ["verbose"],
+        "random_state": ["random_state"],
+        "return_n_iter": ["boolean"],
+        "positive_dict": ["boolean"],
+        "positive_code": ["boolean"],
+        "method_max_iter": [Interval(Integral, 0, None, closed="left")],
+    }
+)
+def dict_learning(
```

Comment:
Sorry for the back and forth but it was finally decided that we can do partial validation for function parameters. It means that the dict should only contain the parameters that are not params of the DictionaryLearning class or that have different constraints.